Source/WebCore/svg/SVGElement.h

```diff
@@ -167,6 +167,8 @@ class SVGElement : public StyledElement, public SVGPropertyOwner {
     SVGConditionalProcessingAttributes& conditionalProcessingAttributes();
     SVGConditionalProcessingAttributes* conditionalProcessingAttributesIfExists() const;
 
+    bool shouldObtainBoundsFromSVGModel() const;
```

Comment:
Hi @nikolaszimmermann, thanks for the review!
I'm confused a little about the relation between `shouldObtainBoundsFromSVGModel` and `shouldObtainBoundsFromBoxModel`.
IIUC, for svg elements, there are 3 scenarios:
1. Without renderer(), no BoundingBox
2. Has an associated [SVG layout box](https://drafts.csswg.org/cssom-view/#svg-layout-box), so get BoundingBox from SVG model.
3. "has an associated CSS layout box", get BoundingBox from Box model.

Except `!renderer()`, for SVG element if scenario 2 is false, does it mean scenario 3 is true? Or could both be true?
And does the rule work for other elements?

Reply:
I will try to remove some of the confusion. In an ideal world, we could simply check if the element is a SVG element and then use SVG getBoundingBox(). However, for the outermost <svg> element, we always need to query the CSS geometry (for various reasons, most prominent one: the placing within compound documents is out of control of SVG itself -> "CSS replaced element").

LBSE complicates things for now, temporarily, because within LBSE we can always rely on the CSS geometry being correct also for SVG elements. However, since we need to run the legacy engine and LBSE in parallel for now, we have these confusing conditions.

I would say, let us stick to the current behavior that there is no bonding box, if there is no renderer (can be improved later if needed). Then all SVG elements would go through the SVG geometry query route, except for LegacyRenderSVGRoot. For LBSE all SVG elements use the CSS geometry route. Does that make sense?

Reply:
Thanks for the explanation! It's much clearer for me. And yes, I think it makes sense to me:) Regarding the function name, could we use `hasAssociatedSVGLayoutBox()` here, and in Element using `shouldObtainBoundsFromSVGModel()` which is matching `shouldObtainBoundsFromBoxModel`?